### ISSUE TITLE
regex capture group, best match method

### ISSUE THREAD

**Issue by wbob** [mallocator/nginx-lua-lang#1]
hi! used your lua script successfully but had initial problems with groking the first regex non-capture group. I changed the expression to my understanding and did so too with the matching loop. I also added some debug suggestions to the readme. Maybe you want to have a look at the changes in the fork if you deem them worthy of a Pullrequest - or you'll uncover my misunderstandings.

**Comment by mallocator** [comment]
Thanks for the feedback and taking some time to look at the script. Just for the future, it'd be easier to ask for a pull request, and then I could annotate the code here with comments. In fact I'm not even sure how to merge your code into this repo without copy and pasting right now.

The change you made to the regex now allows language definitions such as -xz and doesn't require a letter prefix. Now the regex is already not perfect, but I don't see why you have to make it even broader.

The first loop is only there to make sure that all matches are lower case, which I think can be done better further up in the script where the different tables are built / where the input is parsed. 

I'm not sure what the second loop does as the lookup has changed. Did my script have an error there?

But the documentation update seems like a good change, since there's useful extra information and a bunch of errors fixed.

**Comment by wbob** [comment]
I cleaned up my code comments and made a pull request #2 - we can move discussion there if you want to annotate the code.

I'm not sure if there are differences in regex-engines or nginx/lua versions responsible for this, but only after broadening the regex, could I match the full locale (en-EN). See [screenshots](https://imgur.com/a/piEiD) from regex101.com with "gi" flags.
Maybe this was your intention and the script works for you nonetheless.

Why I changed the matching loop: I couldn't make the "if langs[lang[1]] then" match, maybe as the locales where still case-sensitive in the (then sorted) table and changed it to an ordinary for-loop to do an :lower() operation. This is only out of lacking knowledge, as I'm not familiar with the syntax.

I made a local script version of lang.lua and found that only these two parts were lacking for me. With some testdata I get consistent correct results from the nginx version so I thought to share my findings.
